### ISSUE TITLE
Feature Request: Quick Save/Load On-Screen Buttons.

### ISSUE THREAD

**Issue by tgwaste** [rileytestut/Delta#21]

**Comment by BSBussell** [comment]
Could you please provide a description of what you are asking to help clarify what the issue is and how you’d like it resolved?

**Comment by tgwaste** [comment]
I have to ask... Did you read the title?  Can you specific which part is confusing?

**Comment by noah978** [comment]
Actually... this is already possible. Custom controller skins are capable of having buttons to do this. Not sure if anyone had made one yet though.

**Comment by tgwaste** [comment]
Interesting.. Ill look for some documentation.  Are the default ones publicly available to tinker with?

**Comment by noah978** [comment]
I would recommend checking out the Skin Making tutorial I put together [here.](http://github.com/rileytestut/DeltaCore/skins)

It doesn’t include anything about the save state / load state buttons currently, I haven’t gotten around to it. All you need to do is figure out what the names of the buttons are to use as the ID in the JSON file.

You will find all the default skins available for download at the start of the tutorial.

**Comment by tgwaste** [comment]
Thats what I found, but without the IDs for the functions theres nothing to do.

**Comment by noah978** [comment]
Riley said that they exist and I have one of them currently written: menu

So I’m guessing that the others would follow the same scheme: 

* savestate
* loadstate 
* fastforward

You can give it a shot and see if it works.

**Comment by tgwaste** [comment]
The link you gave to your instructions 404s (fyi).
There is no reference to 'thumbstick' in the json file
And when trying to add savestate and loadstate buttons to the json, the app registers the presses but nothing actually happens.

**Comment by noah978** [comment]
I edited the comment so that the link no longer 404s, thumbstick is only in the N64 skin because that’s the only one that normally has it.

And after messaging Riley, he said he forgot to add the final bit of code for custom skins to support save state, fast forward, etc. He also said he’ll try and add that with the next update.

**Comment by tgwaste** [comment]
Bummer.. thanks. Im having a problem with the modded skin where when I save the image (pdf) it somehow activates the background so now I see everything through a semi transparent white background.  I'll have to figure that out.

**Comment by tgwaste** [comment]
was this in 1.1.1 per chance.

**Comment by noah978** [comment]
Unfortunately not, I’d expect in the next minor delta update. This last one was just a small bug fix.

**Comment by JoelStickney** [comment]
I believe this was resolved, but I haven’t found anyone making custom skins with the new buttons yet to test it out.

**Comment by tgwaste** [comment]
I would be happy to try but the last time I did it wasnt there yet.  has it been added?

**Comment by noah978** [comment]
Yes! The new buttons are now available for use and I will be updating the skin tutorial hopefully tonight.

quickSave
quickLoad
fastForward
toggleFastForward

fastForward is the same as what we have with external controllers rn: when you press it, it fast forwards, and when you release, it stops fast forwarding. however, that isn’t rly that convenient for controller skins, so Riley also added toggleFastForward, which just toggles between normal and fast

**Comment by tgwaste** [comment]
tested.  doesn't work at all.  delta 1.1.2

**Comment by tgwaste** [comment]
tested. doesn't work at all. delta 1.1.2
